compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -194,67 +207,77 @@ private class FocusSwitcher<T : Component>(
     private val interopComponent: SwingInteropComponent<T>,
     private val focusManager: FocusManager,
 ) {
-    private val backwardRequester = FocusRequester()
-    private val forwardRequester = FocusRequester()
-    private var isRequesting = false
+    private val backwardTracker = FocusTracker {
```

Comment:
New features of `FocusSwitcher`:
- simple refactoring
- avoiding recursive calls (can happen with empty panels)